### ISSUE TITLE
Is this valid Cloudformation and a cfn-lint bug?

### ISSUE THREAD

**Issue by ispyinternet** [aws-cloudformation/cfn-lint#3632]
**cfn-lint Version **

0.87.3

Syntax:

// add a pseudo element to array (/null) so we can rely on intrinsic length to pop last element we want
 `Id: !Select [ !Length !Split ["/", !Sub "${AWS::StackName}-v1"], !Split [ "/", !Sub "${AWS::StackName}-v1/null" ] ]`

Response:

E0000 did not find expected ',' or ']'

**Comment by ispyinternet** [comment]
Also tried:

```
        Id: 
            Fn::Select:
              [ Fn::Length: 
                !Split ["/", !Sub "${AWS::StackId}-v1/null"] , !Split ["/", !Sub "${AWS::StackId}-v1/null"]
              ]
 ```
 
 Response:
 
E1017 Select index should be an Integer or a function Ref, Fn::FindInMap, or Fn::Select for Resources/EventRule/Properties/Targets/0/Id/Fn::Select/0

**Comment by kddejong** [comment]
Its how you are using the short function syntax `
```
duplication of a tag property (6:32)

 3 |     Type: AWS::SSM::Parameter
 4 |     Properties:
 5 |       Type: String
 6 |       Value: !Select [ !Length !Split ["/", !Sub "${AWS::StackN ...
------------------------------------^
```

`Id: !Select [ !Length {"Fn::Split": ["/", !Sub "${AWS::StackName}-v1"]}, !Split [ "/", !Sub "${AWS::StackName}-v1/null" ] ]`

Looking into the integer issue

**Comment by kddejong** [comment]
In relation to the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-select.html) the question is `Fn::Length` supported

**Comment by ispyinternet** [comment]
Your right, the issue is that !Select index only supports  Ref and Fn::FindInMap functions.

**Comment by kddejong** [comment]
@ispyinternet going to reopen this as the Fn::Length does work when you have the language extension transform added.   I was able to make it work.

This template works.
```yaml
Transform: AWS::LanguageExtensions
Resources:
  Parameter:
    Type: AWS::SSM::Parameter
    Properties:
      Type: String
      Value: !Select [ {"Fn::Length": {"Fn::Split": ["/", !Sub "${AWS::StackName}-v1"]}}, !Split [ "/", !Sub "${AWS::StackName}-v1/null" ] ]
```
